src/main/java/kitchenpos/application/ProductService.java

```diff
@@ -1,34 +1,33 @@
 package kitchenpos.application;
 
-import kitchenpos.dao.ProductDao;
-import kitchenpos.domain.Product;
+import java.util.List;
+
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
-import java.math.BigDecimal;
-import java.util.List;
-import java.util.Objects;
+import kitchenpos.domain.product.Product;
+import kitchenpos.domain.product.ProductRepository;
+import kitchenpos.dto.product.ProductRequest;
+import kitchenpos.dto.product.ProductResponse;
+import kitchenpos.utils.StreamUtils;
 
 @Service
 public class ProductService {
-    private final ProductDao productDao;
+    private final ProductRepository productRepository;
 
-    public ProductService(final ProductDao productDao) {
-        this.productDao = productDao;
+    public ProductService(final ProductRepository productRepository) {
+        this.productRepository = productRepository;
     }
 
     @Transactional
-    public Product create(final Product product) {
-        final BigDecimal price = product.getPrice();
-
-        if (Objects.isNull(price) || price.compareTo(BigDecimal.ZERO) < 0) {
-            throw new IllegalArgumentException();
-        }
-
-        return productDao.save(product);
+    public ProductResponse create(final ProductRequest productRequest) {
+        Product product = Product.of(productRequest.getName(), productRequest.getPrice());
+        return ProductResponse.from(productRepository.save(product));
```

Comment:
개인적으로 request에서 getter를 사용하기 보다는 메시지를 던지는 편을 선호하는데요 🤔 
요청 DTO 에 대한 사용에 있어서 의견이 많이 다를 수 있는 부분이니 참고만 해주세요 🙏 

```suggestion
        Product savedProduct = productRepository.save(request.toProduct());
        return ProductResponse.from(savedProduct);
```

Reply:
Request Model 에 toEntity() 메소드가 있으면 코드 가독성도 높아지고, getter 를 궂이 사용하지 않아도되서 여러모로 장점이 있는 것 같습니다. 이부분도 같이 수정해놓겠습니다!